### ISSUE TITLE
Transmission looks incorrect when no opaque object behind

### ISSUE THREAD

**Issue by hybridherbst** [google/model-viewer#3348]
### Description

Seems transmission is wrongly calculated when there's no other non-transparent object behind a transmissive object.
This can easily be seen with [DragenAttenuation.glb](https://github.com/KhronosGroup/glTF-Sample-Models/blob/master/2.0/DragonAttenuation/glTF-Binary/DragonAttenuation.glb) when viewed from the side:
![20220407-001448_chrome](https://user-images.githubusercontent.com/2693840/162081320-c0b34b08-e01d-488c-91e6-935ff289ccf8.gif)

Compare this with  [Babylon](https://sandbox.babylonjs.com/) or https://gltf-viewer.donmccurdy.com/:
![20220407-001626_chrome](https://user-images.githubusercontent.com/2693840/162081554-bc59fec4-6dba-4fb7-801d-cc1379bfebb3.gif)
![20220407-001655_chrome](https://user-images.githubusercontent.com/2693840/162081597-8a6babaf-4ea2-40db-820d-6f13c2a7410e.gif)

This is also a problem in Augmented Reality: I understand that no refraction can occur (since the camera image isn't actually accessible) but right now objects are simply colorless instead of transmission-colored as expected.

**Comment by elalish** [comment]
That's interesting; can you check if that's specific to the volumetric extension? We had this working properly for the simpler transmission materials before...

**Comment by hybridherbst** [comment]
I think so, same with the [transmission test model ](https://github.com/KhronosGroup/glTF-Sample-Models/tree/0bc2ad7535aba4e16fb1afba1f85fd827bbbea6d/2.0/TransmissionTest/glTF-Binary) which doesn't have the volume extension:
![image](https://user-images.githubusercontent.com/2693840/162083097-cb3845b6-c51c-47b2-a50f-7547b41f7686.png)

**Comment by elalish** [comment]
@mrdoob This looks like a regression; do you have any idea what CLs we should look at?

**Comment by mrdoob** [comment]
Hmm, I've just checked previous versions in https://modelviewer-tester.glitch.me/ and I'm not sure this is a regression 🤔

**Comment by elalish** [comment]
Strange, I remember we checked this in three.js when we were adding support. I'm not sure how MV could break it, but I'll have a look. Does three.js have an example demoing this anywhere?

**Comment by mrdoob** [comment]
If I remember correctly we never managed to have color.
We managed to have some shading showing which was better than nothing: https://github.com/mrdoob/three.js/pull/22425

**Comment by elalish** [comment]
Oh yeah, I guess that was as far as we got. @hybridherbst the difference between MV and those other renderers is they are not rendering to a transparent canvas and allowing that to be stacked in the browser on top of other DOM. We are, but it doesn't (can't) work as well because information is lost. Still, I'm curious if it's possible to do better...

**Comment by hybridherbst** [comment]
Oooh, interesting, so this is a general issue in three when rendering to a transparent canvas?

Would either of you be able to give a quick summary of the related effects that make this hard? I went over the linked issues but they seem to mostly discuss workaorunds and not the underlying issue.

**Comment by mrdoob** [comment]
A very basic summary is that transmission is `material_color * color_behind`.
When the background is transparent there is no `color_behind` to multiply with.

**Comment by elalish** [comment]
The problem is blend functions. For proper transmission, you want background pixels in scene linear light, then attenuate them with color and add in scene linear light from reflections. However, when outputting to a transparent canvas, all your float linear light values (of reflection, since that's all you can calculate without knowing there's a background) get tone-mapped / clamped down to 0-255, and your background is 0-255 as well. Now you get to blend them together after the fact, but you really don't know how they relate anymore. How intense is the background "light" compared to the reflections light? Is the color in your buffer attenuation color that should be multiplied to the background, or reflected light color that should added?

**Comment by elalish** [comment]
Agreed with @mrdoob; it's a pretty cool feature that this is supported at all. It can't be actually correct, but it does a decent job for uncolored (glass-like) transmission.
